core/camel-core-processor/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -880,6 +880,10 @@ private void onSubmitCompletion(final String key, final Exchange exchange) {
 
         // send this exchange
         executorService.execute(() -> {
+            if (exchange.hasOut()) {
```

Comment:
We have a helper method for this

ExchangeHelper.prepareOutToIn(exchange);

Reply:
Thanks @davsclaus and I've repleace to use this helper.